cmake/modules/AddSwift.cmake

```diff
@@ -797,12 +809,18 @@ function(_add_swift_library_single target name)
     set(enable_assertions "${LLVM_ENABLE_ASSERTIONS}")
     set(analyze_code_coverage "${SWIFT_ANALYZE_CODE_COVERAGE}")
   endif()
+
+  if (SWIFTLIB_SINGLE_HOST_LIBRARY)
```

Comment:
The name of the flag is probably misleading.  It means "no matter what the defaults say, also build this library for the host".  So if the defaults say that this library should be built for the host, then this flag won't be set -- which is the case for all compiler libraries.  Maybe this flag should be called something like `FORCE_BUILD_FOR_HOST_SDK`.

The condition you need to check here is `NOT SWIFTLIB_SINGLE_TARGET_LIBRARY `.